### ISSUE TITLE
Scalability test - Sagemaker Notebook

### ISSUE THREAD

**Issue by Ludo171** [awslabs/service-workbench-on-aws#41]
I am currently conducting scalability tests to see if we could safely use AWS-Galileo-Gateway workspaces as materials for classes of 50 to 100 student.

**Errors encountered during the scalability tests**
I triggered the simultaneous creation of 80 small sagemaker notebooks. Only 24 succedeed, 56 failed during their creation.
- No errors reported for any service (I checked lambdas, api gateway, and DynamoDb).
- However, 56 Step Functions failed. I am reporting an error occuring after the step where the workflow-loop-runner ends : 
```
 "error": {
      "stpIndex": 0,
      "stpTmplId": "st-provision-environment",
      "stpTmplVer": 1,
      "wfInstId": "MtJXPd0XZ",
      "wfId": "wf-create-environment",
      "wfVer": 1,
      "code": "ValidationError",
      "time": "2020-06-22T07:57:31.310Z",
      "requestId": "a78dc82a-59c0-4423-aece-996d7c01b9ae",
      "statusCode": 400,
      "retryable": false,
      "retryDelay": 85.8109820630679,
      "msg": "1 validation error detected: Value null at 'stackName' failed to satisfy constraint: Member must not be null",
      "stack": "ValidationError: 1 validation error detected: Value null at 'stackName' failed to satisfy constraint: Member must not be null\n    at Request.extractError (/var/runtime/node_modules/aws-sdk/lib/protocol/query.js:50:29)\n    at Request.callListeners (/var/runtime/node_modules/aws-sdk/lib/sequential_exe"
}
```

![step-function-failure](https://user-images.githubusercontent.com/33128425/85286433-f19c5200-b492-11ea-99cb-83dae5cbb8ad.png)

![step-function-failure-2](https://user-images.githubusercontent.com/33128425/85286723-71c2b780-b493-11ea-9172-9b1e63729d29.png)

**Suggestions about the potential cause of this issue ?**
?

**Reproduce the same test**
Using the node scripts in `/scripts/load-test-workspaces/` that I had to update in PR #42 
- Have a look at the [Readme file](./scripts/load-test-workspaces/README.md) to set up the test on your environment in 2 min. 

Once you're set, run :
- `cd ./scripts/load-test/`
- `node load-test-workspaces create --count 80 --platform sagemaker-1 --size sagemaker__small` _(this will make 80 simultaneous calls to the API endpoint that create workspaces, decrease `--count` as you want.)_

**Consumer account's limits are not limiting.**
I made sure that the limits of the consumer account used for this test are not limiting anything (limit for Number of Running Notebook Instances is 200, and we also increased the EC2 limit to 392 vCPUs). In addition to this I checked that we do not have much resources already occupied in this account.

**Comment by kennethgillen** [comment]
@Ludo171 did you end up using this for teaching? If so, how did it go?

**Comment by Ludo171** [comment]
Hello @Kenny,
Sorry for the delay of my answer.

In fact, we performed successfully a class with the proof-of-concept
version of Service Workbench (in beginning of 2020).
We used it to provide a Sagemaker environment to each student. Globally, it
worked fine. We stumbled on issues of management of R packages (because we
use R a lot at our lab) when the instances were stopped then re-started
another day. The issue was specific to using R on Sagemaker which is why we
promoted the implementation of new workspace types consisting of EC2
running RStudio servers. We had no issue with the python libraries on
Sagemaker.

Since then, Service Workbench has been refactored into this open-source
solution, and has become more stable. We are still giving feedback to the
development team to make it fit even better the needs of research &
teaching teams.
